ktor-server/ktor-server-core/jvm/src/io/ktor/features/ContentNegotiation.kt

```diff
@@ -110,6 +110,8 @@ public class ContentNegotiation(
                     proceed()
                 } catch (e: UnsupportedMediaTypeException) {
                     call.respond(HttpStatusCode.UnsupportedMediaType)
+                } catch (e: ContentTransformationException) {
+                    call.respond(HttpStatusCode.BadRequest, e.message ?: "Error transforming content")
```

Comment:
The problem is that this may happen on sending...

Reply:
when will it happen on sending? is the same code used for outgoing json? where would be a better place to catch this and return the correct HTTP status?